firewood/examples/insert.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use clap::Parser;
+use std::{error::Error, ops::RangeInclusive, sync::Arc, time::Instant};
+
+use firewood::{
+    db::{Batch, BatchOp, Db, DbConfig},
+    v2::api::{Db as DbApi, Proposal},
+};
+use rand::{distributions::Alphanumeric, Rng};
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(short, long, default_value = "1-64", value_parser = string_to_range)]
+    keylen: RangeInclusive<usize>,
+    #[arg(short, long, default_value = "32", value_parser = string_to_range)]
+    datalen: RangeInclusive<usize>,
+    #[arg(short, long, default_value_t = 1)]
+    batch_keys: usize,
+    #[arg(short, long, default_value_t = 100)]
+    inserts: usize,
+}
+
+fn string_to_range(input: &str) -> Result<RangeInclusive<usize>, Box<dyn Error + Sync + Send>> {
+    //<usize as std::str::FromStr>::Err> {
+    let parts: Vec<&str> = input.split('-').collect();
+    match parts.len() {
+        1 => Ok(input.parse()?..=input.parse()?),
+        2 => Ok(parts[0].parse()?..=parts[1].parse()?),
+        _ => Err("Too many dashes in input string".into()),
+    }
+}
+
+/// cargo run --release --example insert
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let cfg = DbConfig::builder().truncate(true).build();
+
+    let args = Args::parse();
+
+    let db = tokio::task::spawn_blocking(move || {
+        Db::new("rev_db", &cfg).expect("db initiation should succeed")
+    })
+    .await
+    .unwrap();
+
+    let keys = args.batch_keys;
+    let start = Instant::now();
+    for _ in 0..args.inserts {
+        let keylen = rand::thread_rng().gen_range(args.keylen.clone());
+        let datalen = rand::thread_rng().gen_range(args.datalen.clone());
```

Comment:
I have no idea how quick these actually are, but if you're communicating between threads, it's not trivial, right? You should precompute all the lengths and iterate over the pairs.

```rust
let key_and_data_lengths = (0..args.inserts)
    .map(|_| (
        rand::thread_rng().gen_range(args.keylen.clone()),
        rand::thread_rng().gen_range(args.datalen.clone()),
    ))
    .collect::<Vec<(usize, size)>>();
    
let start = Instant::now();
    
for (keylen, datalen) in key_and_data_lengths {
  // ...
```

Reply:
actual change request

Reply:
The goal here wasn't an actual benchmark. If I wanted that, I'd have used the benchmarking stuff that's built in that supports these kinds of assumptions. That's why this is in the examples directory.

Your point that it probably shouldn't count the time for the generation of the keys is valid, though. I'm more tempted to just removing the timings and call it a true example though. Thoughts?

Reply:
Maybe just add a comment about the time? 

Can be done in a follow up